runtime/lua/vim/lsp/util.lua

```diff
@@ -1635,41 +1667,69 @@ function M.try_trim_markdown_code_blocks(lines)
   return 'markdown'
 end
 
-local str_utfindex = vim.str_utfindex
 ---@private
-local function make_position_param()
-  local row, col = unpack(api.nvim_win_get_cursor(0))
+---@param window (optional, number): window handle or 0 for current, defaults to current
+---@param offset_encoding string utf-8|utf-16|utf-32|nil defaults to utf-16
+local function make_position_param(window, offset_encoding)
```

Comment:
What do you think of changing the default of `offset_encoding` from `utf-16` to instead call call `_get_offset_encoding` ? (Same for all other `make_xy_param(s)` functions)

This would have the advantage that the fix would propagate to plugins as well and we could omit the changes in `buf.lua`

Reply:
This is a great idea, my only concern is that this makes things a bit opaque on the plugin side. That is, if you're persistently storing these parameters -- as we do for our "pins" in lean.nvim -- you don't know what encoding they came from. It would probably be okay to add the encoding as a second return value, right?

Reply:
Just pushed, let me know what you think.

Reply:
> you don't know what encoding they came from. It would probably be okay to add the encoding as a second return value, right?

I think if the caller needs to know the encoding they should set it explicitly - which is still allowed. I'd prefer it if the method didn't return it as part of the result.

Reply:
This would kind of force use to either use the private function `_get_offset_encoding` or have some ugly logic for independently determining the offset encoding to pass in. I don't really see an issue if the encoding is an inherent property of the position params? IMO the protocol should even be extended to include it as its own (optional) field with the params themselves.

Reply:
Ok so the issue we have is that in our plugin we're also using these position params for UI features, specifically for creating extmarks. I suppose an easier and cleaner fix would be to just have another copy of these params UTF-8 encoded for this purpose alone. So will revert the second return value.